### PR TITLE
refactor: replacing left and right to start and end

### LINKED ISSUE #2215: Make `start` and `end` the default
### Description

Currently, Yamada UI components default to `left` and `right`. Replace them with `start` and `end`. Also, fix `props` that do not support `start` and `end`.

### Problem Statement/Justification

Expressions such as `left` and `right` change their meaning when the element is rotated.

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by mokun632**
ref: #2215 

## Description
Expressions such as left and right have the nuance of elements rotating, so we replaced them with start and end.
The main replacement is for props such as leftIcon and rightIcon.
If there are any other props that should be replaced, I will work on them in a separate PR.

### CODE REVIEW

**Comment by mokun632** [comment]
It seems that the test is failing in a place that is not the part that was changed this time.
The same part is also failing in the base branch.
Should I fix this in this PR?

File: packages/components/button/src/button.tsx [hunk 1]
```diff
@@ -47,11 +47,11 @@ type ButtonOptions = {
   /**
    * The icon to display at the left side of the button.
```
Comment (hirotomoyamada) [inline packages/components/button/src/button.tsx:48]: Description needs to be changed.

File: packages/components/tag/src/tag.tsx [hunk 2]
```diff
@@ -15,11 +15,11 @@ type TagOptions = {
   /**
    * Icon to be displayed to the left of the tag.
```
Comment (hirotomoyamada) [inline packages/components/tag/src/tag.tsx:16]: Description needs to be changed.

**Comment by hirotomoyamada** [comment]
@mokun632 

There are many things to change, such as the `InputLeftElement` and the `placement` of the `Indicator`.

**Comment by hirotomoyamada** [comment]
@mokun632 

Even if we add `start` and `end`, we want to keep `left` and `right`.We want them to coexist.

**Comment by mokun632** [comment]
@hirotomoyamada 
>There are many things to change, such as the InputLeftElement and the placement of the Indicator.

I think `left` and `right` should be used for items that indicate position, such as `placement`.
For items that indicate start or end, I think `start` and `end` are better.
As for `InputLeftElement`, I will fix it in another PR.

**Comment by hirotomoyamada** [comment]
@mokun632 

Difference.Replace all with `start` and `end`, but keep `right` and `left`.

**Comment by mokun632** [comment]
@hirotomoyamada 
Am I correct in assuming that you want left and right to remain for backward compatibility?

**Comment by hirotomoyamada** [comment]
@mokun632 

Yes, it is.

**Comment by mokun632** [comment]
@hirotomoyamada 
Please tell me, thank you!

> Replace them with start and end. 

However, the problem is explained above.
It would be helpful if you could explain the context and content more clearly next time!

**Comment by hirotomoyamada** [comment]
@mokun632 

Hi, let's talk on Discord once.

**Comment by mokun632** [comment]
@hirotomoyamada 
Please review!
`start` and `end` now take precedence when `start`,`end` and `left`,`right ` are used at the same time.
Also, I think we need to test that `start`,`end` is prioritized, but more would bloat the PR, so I'll put it up in an issue when this PR is merged.

File: packages/components/button/src/button.tsx [hunk 3]
```diff
@@ -45,11 +45,23 @@ type ButtonOptions = {
    */
   isDisabled?: boolean
   /**
-   * The icon to display at the left side of the button.
+   * The icon to display at the start side of the button.
+   */
+  startIcon?: ReactElement
+  /**
+   * Icon to be displayed to the left of the tag.
```
Comment (hirotomoyamada) [inline packages/components/button/src/button.tsx:52]: Change

```
Icon to be displayed to the left of the tag.
```

to

```
The icon to display at the left side of the button.
```

File: packages/components/button/src/button.tsx [hunk 4]
```diff
@@ -45,11 +45,23 @@ type ButtonOptions = {
...
+   * @deprecated Use `startIcon` instead.
    */
   leftIcon?: ReactElement
   /**
-   * The icon to display at the right side of the button.
+   * The icon to display at the end side of the button.
+   */
+  endIcon?: ReactElement
+  /**
+   * Icon to be displayed to the right of the tag.
```
Comment (hirotomoyamada) [inline packages/components/button/src/button.tsx:62]: Change

```
Icon to be displayed to the right of the tag.
```

to

```
The icon to display at the right side of the button.
```

File: packages/components/button/src/button.tsx [hunk 5]
```diff
@@ -207,16 +223,25 @@ const Loading: FC<
...
+    "startIcon" | "leftIcon" | "endIcon" | "rightIcon" | "children"
+  >
+> = ({
+  startIcon: _startIcon,
   leftIcon,
+  endIcon: _endIcon,
   rightIcon,
   children,
 }) => {
+  const startIcon = _startIcon ?? leftIcon
```
Comment (hirotomoyamada) [inline packages/components/button/src/button.tsx:238]: Change

```
const startIcon = _startIcon ?? leftIcon
const endIcon = _endIcon ?? rightIcon
```

to

```
startIcon ??= leftIcon
endIcon ??= rightIcon
```

File: packages/components/button/src/icon-button.tsx [hunk 6]
```diff
@@ -13,7 +13,7 @@ type IconButtonOptions = {
 
 export type IconButtonProps = Omit<
   ButtonProps,
-  "leftIcon" | "rightIcon" | "loadingIcon" | "loadingText" | "loadingPlacement"
+  "startIcon" | "endIcon" | "loadingIcon" | "loadingText" | "loadingPlacement"
```
Comment (hirotomoyamada) [inline packages/components/button/src/icon-button.tsx:16]: Add `leftIcon` and `rightIcon`.

File: packages/components/button/src/button.tsx [hunk 7]
```diff
@@ -45,11 +45,23 @@ type ButtonOptions = {
...
   isDisabled?: boolean
   /**
-   * The icon to display at the left side of the button.
+   * The icon to display at the start side of the button.
+   */
+  startIcon?: ReactElement
+  /**
+   * Icon to be displayed to the left of the tag.
+   * If specified at the same time as `startIcon`, `startIcon` takes precedence.
+   * @deprecated Use `startIcon` instead.
```
Comment (hirotomoyamada) [inline packages/components/button/src/button.tsx:54]: Change

```
   * If specified at the same time as `startIcon`, `startIcon` takes precedence.
   * @deprecated Use `startIcon` instead.
```

to

```
   * If specified at the same time as `startIcon`, `startIcon` takes precedence.
   * 
   * @deprecated Use `startIcon` instead.
```

File: packages/components/button/src/button.tsx [hunk 8]
```diff
@@ -45,11 +45,23 @@ type ButtonOptions = {
...
   leftIcon?: ReactElement
   /**
-   * The icon to display at the right side of the button.
+   * The icon to display at the end side of the button.
+   */
+  endIcon?: ReactElement
+  /**
+   * Icon to be displayed to the right of the tag.
+   * If specified at the same time as `endIcon`, `endIcon` takes precedence.
+   * @deprecated Use `endIcon` instead.
```
Comment (hirotomoyamada) [inline packages/components/button/src/button.tsx:64]: Change

```
   * If specified at the same time as `endIcon`, `endIcon` takes precedence.
   * @deprecated Use `endIcon` instead.
```

to

```
   * If specified at the same time as `endIcon`, `endIcon` takes precedence.
   * 
   * @deprecated Use `endIcon` instead.
```

File: packages/components/tag/src/tag.tsx [hunk 9]
```diff
@@ -12,12 +12,24 @@ import type { FC, HTMLAttributes, MouseEventHandler, ReactElement } from "react"
...
 
 type TagOptions = {
+  /**
+   * Icon to be displayed to the start of the tag.
+   */
+  startIcon?: ReactElement
   /**
    * Icon to be displayed to the left of the tag.
+   * If specified at the same time as `startIcon`, `startIcon` takes precedence.
+   * @deprecated Use `startIcon` instead.
```
Comment (hirotomoyamada) [inline packages/components/tag/src/tag.tsx:22]: Change

```
   * If specified at the same time as `startIcon`, `startIcon` takes precedence.
   * @deprecated Use `startIcon` instead.
```

to

```
   * If specified at the same time as `startIcon`, `startIcon` takes precedence.
   * 
   * @deprecated Use `startIcon` instead.
```

File: packages/components/tag/src/tag.tsx [hunk 10]
```diff
@@ -12,12 +12,24 @@ import type { FC, HTMLAttributes, MouseEventHandler, ReactElement } from "react"
...
    */
   leftIcon?: ReactElement
+  /**
+   * Icon to be displayed to the end of the tag.
+   */
+  endIcon?: ReactElement
   /**
    * Icon to be displayed to the right of the tag.
+   * If specified at the same time as `endIcon`, `endIcon` takes precedence.
+   * @deprecated Use `endIcon` instead.
```
Comment (hirotomoyamada) [inline packages/components/tag/src/tag.tsx:32]: Change

```
   * If specified at the same time as `endIcon`, `endIcon` takes precedence.
   * @deprecated Use `endIcon` instead.
```

to

```
   * If specified at the same time as `endIcon`, `endIcon` takes precedence.
   * 
   * @deprecated Use `endIcon` instead.
```

**Comment by mokun632** [comment]
@hirotomoyamada 
I fixed comments.
Please review again!

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**